web-bundle/src/main/java/com/graphhopper/resources/RouteResource.java

```diff
@@ -233,7 +233,7 @@ private void translateTurnCostsParamToEdgeBased(GHRequest request, MultivaluedMa
             if (turnCosts.size() != 1) {
                 throw new IllegalArgumentException("You may only specify the turn_costs parameter once");
             }
-            request.getHints().put(EDGE_BASED, turnCosts.get(0));
+            request.putHint(EDGE_BASED, turnCosts.get(0));
```

Comment:
Here we need `Helper.toObject(turnCosts.get(0))`, as mentioned here (in case you missed this): https://github.com/graphhopper/graphhopper/pull/1956#discussion_r394858559